reconcile/openshift_resources_base.py

```diff
@@ -229,10 +229,11 @@ def lookup_github_file_content(repo, path, ref, tvars=None):
     return c.decode("utf-8")
 
 
-def lookup_graphql_query_results(query: str) -> list[Any]:
+def lookup_graphql_query_results(query: str, kwargs: dict[str, Any] = {}) -> list[Any]:
```

Comment:
That mutable default is going to be problematic, I'd be surprised if the linter let it pass.
```suggestion
def lookup_graphql_query_results(query: str, kwargs: Optional[dict[str, Any]]= None) -> list[Any]:
    if kwargs is None:
        kwargs = {}
```
Also, kwargs is usually reserved to `def foo(*args, **kwargs)` so that we could call this as `lookup_graphql_query_results('{...}', placeholder1='hello', placeholder2='world')`

That way your `render` would be:

```python
rendered_resource = jinja2.Template(resource).render(**kwargs)
```

Also, in that case I wouldn't bother typing `**kwargs`.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/2277/commits/4cb9542b99fc292d32810e7fc02cc09ba1c80c0a